app/models/chapter.rb

```diff
@@ -45,4 +45,36 @@ def display_with_mlid
   def full_mlid
     "#{organization.mlid}-#{mlid}"
   end
+
+  def delete_chapter_and_dependents
+    transaction do
+      self.deleted_at = Time.zone.now
+
+      # rubocop:disable Rails/SkipsModelValidations
+      groups_to_update = Group.includes(:chapter).where(chapter_id: id, deleted_at: nil)
+      group_ids = groups_to_update.pluck(:id)
+
+      groups_to_update.update_all(deleted_at:)
+      Student.includes(:group).where(groups: group_ids, deleted_at: nil).update_all(deleted_at:)
```

Comment:
no need to call `includes` here. It will join with the `groups` table allowing you to call `where(groups:...` but since student table has the `group_id` field we can just do it like this:

`Student.where(group_id: group_ids, deleted_at: nil).update_all(deleted_at:)`